### ISSUE TITLE
Mission transition problem!

### ISSUE THREAD

**Issue by yigit-sen** [BG-Software-LLC/SuperiorSkyblock2#638]
### Minecraft's Version

1.16.5 and purpur.jar

### Plugin's Version

SuperiorSkyblock2-1.8-b324.jar

### Describe the bug

First of all, the error that I will explain occurs in some players. 

My missions progress step by step and after the player completes the first mission, instead of going to the second mission automatically, they start the first mission again automatically.

i write /stop. When I stop the server completely, the player automatically switches to the second mission, but this time it automatically restarts when he finishes that mission and continues like this.

### To Reproduce

the error that I will explain occurs in some players. 

### Additional Information

mission/_data.yml save problem? or required_missions problem?

I'm throwing two tasks as an example
https://paste.gg/p/anonymous/648bd18c2bbb4570baf1de3e73eaa6ee

Thank you!

**Comment by OmerBenGera** [comment]
try using only english characters, it might fix the issue.
Also, you can't complete the same mission again, as once you finish a mission, it's finished.

**Comment by SOLJAOVLS12** [comment]
Im having this same issue players complete there mission and once they click it and reopen the menu it still shows they can claim again and cant move onto the next mission till i restart the server then all missions save

**Comment by OmerBenGera** [comment]
> 
> 
> Im having this same issue players complete there mission and once they click it and reopen the menu it still shows they can claim again and cant move onto the next mission till i restart the server then all missions save

Does it happen with default files as well?

**Comment by OmerBenGera** [comment]
Take a video please.
Also, make sure no errors occur.

**Comment by SOLJAOVLS12** [comment]
Have more players test or get alts on as it dosnt affect every player it's real strange it's like 10% of my playerbase that it happens to

**Comment by AldenSH** [comment]
Will old data be deleted? Will we have to do the quests from scratch?

**Comment by yigit-sen** [comment]
I couldn't delete.

**Comment by yigit-sen** [comment]
Hey, i found something. Same player's data is in different missions.

**Comment by OmerBenGera** [comment]
> Hey, i found something. Same player's data is in different missions.

Mind explaining what you meant here?

**Comment by yigit-sen** [comment]
Same player's data is in different missions in /missions/_data.yml

**Comment by OmerBenGera** [comment]
> 
> 
> Same player's data is in different missions in /missions/_data.yml

Show me an example please.

**Comment by yigit-sen** [comment]
UUID of player enesakif7 is 319ac20d-5b8b-3610-9485-f44debe35516
enesakif7:  For example this player has no problem and has uuids in _data.yml.
![image](https://user-images.githubusercontent.com/60177116/140569806-ecbee16b-9a82-4cc6-8e00-4464f834f925.png)

UUID of player The_Osman is 38481b0e-e08b-3475-a735-52adfd37a1f7
This man has a duty problem. _data.yml does not have uuids. 
![image](https://user-images.githubusercontent.com/60177116/140569995-b6bdf022-6268-495f-92e0-03ef376ca509.png)


In short, users who have problems are not registered in _data.yml.

**Comment by OmerBenGera** [comment]
> 
> 
> UUID of player enesakif7 is 319ac20d-5b8b-3610-9485-f44debe35516 enesakif7: For example this player has no problem and has uuids in _data.yml. ![image](https://user-images.githubusercontent.com/60177116/140569806-ecbee16b-9a82-4cc6-8e00-4464f834f925.png)
> 
> UUID of player The_Osman is 38481b0e-e08b-3475-a735-52adfd37a1f7 This man has a duty problem. _data.yml does not have uuids. ![image](https://user-images.githubusercontent.com/60177116/140569995-b6bdf022-6268-495f-92e0-03ef376ca509.png)
> 
> In short, users who have problems are not registered in _data.yml.

Data is only saved when server stops.
Are you sure that what causes the issue?

**Comment by yigit-sen** [comment]
It is most likely due to this, the player has been playing for a long time, even his data has not been kept. After finishing the task, he may have had a problem while transferring his data to another task.

**Comment by OmerBenGera** [comment]
@yigit-sen If you add him manually to the file when the server is stopped, does it fix the issue for him?

**Comment by OmerBenGera** [comment]
Have you tried it? @yigit-sen

**Comment by OmerBenGera** [comment]
I finally found what caused it, should be fixed in latest build.
